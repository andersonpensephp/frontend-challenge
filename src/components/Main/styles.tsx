'use client'

import styled from 'styled-components'

export const Wrapper = styled.main`
  color: #fff;
  background: blue;
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;
  height: 100%;
`

export const Title = styled.h1`
  font-size: 2.5rem;
  margin: 10px 0;
`

export const Description = styled.p`
  font-size: 1.5rem;
  margin: 10px 0;
`
