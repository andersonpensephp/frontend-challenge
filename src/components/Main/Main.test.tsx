import { render, screen } from '@testing-library/react'

import Main from './Main'
describe('<Main />', () => {
  it('should...', () => {
    const { container } = render(<Main />)
    expect(
      screen.getByRole('heading', { name: /Reactjs/i })
    ).toBeInTheDocument()
    //Utilizar snapshot somente quando você quer garantir uma estrutura no código
    expect(container.firstChild).toMatchSnapshot()
  })
})
