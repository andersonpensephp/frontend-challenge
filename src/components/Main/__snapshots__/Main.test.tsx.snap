// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> should... 1`] = `
.c0 {
  color: #fff;
  background: blue;
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;
  height: 100%;
}

.c1 {
  font-size: 2.5rem;
  margin: 10px 0;
}

.c2 {
  font-size: 1.5rem;
  margin: 10px 0;
}

<main
  class="c0"
>
  <h1
    class="c1"
  >
    Reactjs
  </h1>
  <p
    class="c2"
  >
    Descrição
  </p>
</main>
`;
